### PR TITLE
[SPARK-10427][SQL]Respect -S option for spark-sql

### PR DESCRIPTION
**Pull request by zhichao-li**
"-S" option is ignored by the conf setting logic at the moment, we need to pick up this to complete the "silent" function.  

Before:
lizhichao@lizhichao-desktop:~/repo/spark/bin$ ./spark-sql -S -e "show databases"
SET hive.support.sql11.reserved.keywords=false
SET spark.sql.hive.version=1.2.1
SET spark.sql.hive.version=1.2.1
OK
default

After:
lizhichao@lizhichao-desktop:~/repo/spark/bin$ ./spark-sql -S -e "show databases"  
default

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala [hunk 1]
```diff
@@ -174,8 +174,6 @@ private[hive] class IsolatedClientLoader(
 
   // Pre-reflective instantiation setup.
   logDebug("Initializing the logger to avoid disaster...")
-  Thread.currentThread.setContextClassLoader(classLoader)
-
```
Comment (zhichao-li) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala]: It is redundant here since the `contexClasstloader` would be reset correctly within the constructor of `ClientWrapper`, and setting the contextClassLoader too early here would prevent us to retrieve the other classLoader within the same thread which having `SessionState` containing the user input option.

**Comment by zhichao-li** [comment]
ping @chenghao-intel @yhuai @liancheng

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala [hunk 2]
```diff
@@ -174,8 +174,6 @@ private[hive] class IsolatedClientLoader(
 
   // Pre-reflective instantiation setup.
   logDebug("Initializing the logger to avoid disaster...")
-  Thread.currentThread.setContextClassLoader(classLoader)
```
Comment (chenghao-intel) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala]: This probably causes ClassNotFoundException or ClassCastException if we removed this line, as we do need to load the specified version of Hive Metastore jars and its dependencies.

**Comment by andrewor14** [comment]
@zhichao-li can you rebase to master?
@liancheng please have a look. Does this still apply?

**Comment by zhichao-li** [comment]
@andrewor14  @liancheng , These options like "-S" "-v" still be discarded since they are all stored within SessionState which would be reconstructed by HiveContext while creating ClientWrapper.

The logic flow kind of like:
SessionState  is stored in a static field. 
1)sparkClassLoader(SparkSQLCLIDriver) -> SessionState (containing user parameters)
2)sparkClassLoader(executationHive) -> reconstruct SessionState
3)MutableURLClassLoader(metaDataHive) -> reconstruct SessionState

Somehow we need to pass the parameters from sparkClassLoader to MutableURLClassLoader, my approach is to defined a new class which share the same root loader. 

It's kind of hack here, not sure it's worth a try for such tiny issue.

**Comment by liancheng** [comment]
@zhichao-li My feeling is that it's probably not worth so much effort for such a tiny feature...

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 3]
```diff
@@ -263,6 +263,7 @@ private[hive] object SparkSQLCLIDriver extends Logging {
 
 }
 
+
```
Comment (chenghao-intel) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala]: revert the change

**Comment by chenghao-intel** [comment]
@liancheng some of our customer complaints that Spark SQL produces too many logs in their cronjob when immigrate the existing scripts from the legacy product. As Hive will suppress the log when given the option `-s`.

**Comment by zhichao-li** [comment]
retest this please.

**Comment by zhichao-li** [comment]
retest this please.

**Comment by zhichao-li** [comment]
retest this please.

**Comment by rxin** [comment]
Thanks for the pull request. I'm going through a list of pull requests to cut them down since the sheer number is breaking some of the tooling we have. Due to lack of activity on this pull request, I'm going to push a commit to close it. Feel free to reopen it or create a new one.
